motr/setup.c

```diff
@@ -1545,35 +1545,88 @@ static void cs_part_domain_setup(struct m0_reqh_context *rctx)
 		part_cfg->bpc_location =
 			(char*)cs_storage_partdom_location_gen(sdev->sd_filename);
 		part_cfg->bpc_dom_key = sdev->sd_dev_idx;
-		/** seg0 configuration */
-		part_cfg->bpc_part_mode_seg0 = true;
-		rctx->rc_be_seg0_path = sdev->sd_filename;
-		proposed_chunk_size = sdev->sd_size / def_dev_chunk_count;
-		part_cfg->bpc_chunk_size_in_bits =
-			align_chunk_size(proposed_chunk_size);
-		part_cfg->bpc_total_chunk_count =
-			sdev->sd_size >> part_cfg->bpc_chunk_size_in_bits;
-
-		/** seg1 configuration */
-		m0_bcount_t meta_size = (sdev->sd_size) / 10;
-		part_cfg->bpc_part_mode_seg1 = true;
-		rctx->rc_be_seg_path = sdev->sd_filename;
-		if(rctx->rc_be_seg_size > meta_size){
+		if (rctx->rc_be_seg_path == NULL &&
```

Comment:
cs_conf_get_parition_dev would need modification to indicate how many data discs are mentioned; to check if single meta daisc specified